corehq/apps/app_manager/static/app_manager/js/report-module.js

```diff
@@ -1,17 +1,108 @@
 
 var ReportModule = (function () {
 
-    function ReportConfig(report_id, display, availableReportIds, language) {
+    function Config(dict) {
+        var self = this;
+
+        dict = dict || {};
+
+        self.keyValuePairs = ko.observableArray();
+        for (var key in dict) {
+            self.keyValuePairs.push([ko.observable(key), ko.observable(dict[key])]);
+        }
+
+        self.addConfig = function() {
+            self.keyValuePairs.push([ko.observable(''), ko.observable('')]);
```

Comment:
KO doesn't come with observable dictionaries, so this is my implementation for purposes of this UI.  Strictly speaking it's not a conventional dict since it can have multiple key-value pairs with the same key, but you're allowed to put the same key multiple times in the UI anyway, and I'm trusting that users are going to be smart enough to realize that would result in not-clearly-defined behavior (I'm treating this as an R&D feature anyway - @czue If I'm assuming too much from the users, let me know)